### PR TITLE
Add OSS caching

### LINKED ISSUE #11: [FEATURE] Save OSS Index results to cache
* **What are you trying to do?**

Generally CVEs/etc... aren't applied at the speed of light, so it's more or less ok to cache results from OSS Index for a period of time. This will prevent users from running into rate limiting on the service, and will also speed up the application in general.

* **What feature or behavior is this required for?**
- Prevents rate limiting
- Speed up application

* **How could we solve this issue? (Not knowing is okay!)**
I have literally no idea what to do here, so anyone with extensive R knowledge could likely lead us towards a key-value store of some sort where we could persist OSS Index responses. We'd cache them by coordinate with a TTL of 12 hours, more or less.

* **Anything else?**

Have a BLAST!

cc @bhamail / @DarthHater / @brittanybelle / @adrianpowell / @csgillespie

### PR DESCRIPTION
**Pull request by csgillespie**
Add's package caching (fixes #11)

 * In R4.+, it gives the facility to create a user directory across OS
 * Create a cache file and use as appropriate
 * Add a timestamp to packages for a maximum of 12 hours caching (per package)
 * Package can be cleared using `clean_cache()`

-----

Other changes

 * Deleted a test that uses HTTP mock. This test didn't offer anything new and was a bit of a pain
 * Make the audit-files function consistent

cc @bhamail / @DarthHater / @brittanybelle / @adrianpowell / @csgillespie

### CODE REVIEW

File: R/cache.R [hunk 1]
```diff
@@ -0,0 +1,54 @@
+ensure_cache = function() {
+  dir = tools::R_user_dir("oysteR", which = "cache")
+  path = file.path(dir, "cached-deps.rds")
```
Comment (bhamail) [inline R/cache.R:3]: minor: Is there a way to declare this filename as a shared "constant"? The the string would not be duplicated here and in `remove_cache()`. It's already pretty safe, as it is encapsulated in `cache.R`, but thought I'd ask and learn. :)

File: R/cache.R [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+#' Remove cache
+#'
+#' The OSS cache is located at `tools::R_user_dir("oysteR", which = "cache")`.
+#' The function `R_user_dir()` is only available for R >= 4.0.0.
+#' Packages are cached for 24 hours, then refreshed at the next audit
+#' @export
+remove_cache = function() {
+  dir = tools::R_user_dir("oysteR", which = "cache")
+  path = file.path(dir, "cached-deps.rds")
```
Comment (bhamail) [inline R/cache.R:52]: see my prior silly question

**Review by bhamail — APPROVED** [review]
LGTM, but I am very light on clues.

File: R/cache.R (hunk 1 above)
Comment (csgillespie) [reply R/cache.R:3]: No constants in R. But I've created a single get method

**Pull request merged**